package/MDAnalysis/lib/formats/libdcd.pyx

```diff
@@ -590,14 +539,22 @@ cdef class DCDFile:
                                'in mode "r"'.format('self.mode'))
         if self.n_frames == 0:
             raise IOError("opened empty file. No frames are saved")
-
-        cdef np.ndarray xyz = np.empty((self.natoms, self.ndims), dtype=FLOAT, order='F')
-        cdef np.ndarray unitcell = np.empty(6, dtype=DOUBLE)
+        cdef np.npy_intp[2] dims 
+        dims[0] = self.natoms
+        dims[1] = self.ndims
+        # note use of fortran flag (1)
+        cdef np.ndarray[FLOAT_T, ndim=2] xyz = np.PyArray_EMPTY(2, dims, np.NPY_FLOAT32, 1)
```

Comment:
allocate this once per reader as a private scratch buffer

Reply:
Done